devito/passes/iet/definitions.py

```diff
@@ -219,6 +219,14 @@ def _alloc_object_array_on_low_lat_mem(self, site, obj, storage):
 
         storage.update(obj, site, allocs=decl)
 
+    def _alloc_petsc_array_on_low_lat_mem(self, site, obj, storage):
```

Comment:
Looks exactly the same as `_alloc_object_array_on_low_lat_mem` robably only needs 
`elif i.is_ObjectArray or i.is_PETScArray:` below